artemis-server/src/main/java/org/apache/activemq/artemis/core/server/balancing/policies/PolicyFactoryResolver.java

```diff
@@ -56,19 +59,15 @@ private void loadPolicyFactories() {
          PolicyFactory.class, BrokerBalancer.class.getClassLoader());
 
       for (PolicyFactory policyFactory : serviceLoader) {
-         registerPolicyFactory(policyFactory);
+         policyFactories.put(keyFromClassName(policyFactory.getClass().getName()), policyFactory);
       }
    }
 
-   public void registerPolicyFactory(PolicyFactory policyFactory) {
-      for (String policyName : policyFactory.getSupportedPolicies()) {
-         policyFactories.put(policyName, policyFactory);
-      }
+   public void registerPolicyFactory(String name, PolicyFactory policyFactory) {
+      policyFactories.put(name, policyFactory);
    }
 
-   public void unregisterPolicyFactory(PolicyFactory policyFactory) {
-      for (String policyName : policyFactory.getSupportedPolicies()) {
-         policyFactories.remove(policyName, policyFactory);
-      }
+   String keyFromClassName(String name) {
+      return name.substring(0, name.indexOf("PolicyFactory"));
```

Comment:
Is it possible to avoid full/partial class names in the broker.xml?

Reply:
I think it makes sense to use part of the newly provided class to provide a name to identify the feature. That class wil have to implement a specific interface so it will already be tied to a class. But in this case, the name gives the actual intent of the class. I don't see need for any indirection.